aten/src/ATen/native/NativeFunctions.cpp

```diff
@@ -27,6 +27,69 @@ std::vector<Tensor> split(const Tensor& self, int64_t split_size, int64_t dim) {
   return splits;
 }
 
+Tensor slice(const Tensor& self, int64_t start, int64_t end, int64_t step, int64_t dim) {
+  dim = maybe_wrap_dim(dim, self.dim());
+  auto sizes = std::vector<int64_t>(self.sizes());
+  auto strides = std::vector<int64_t>(self.strides());
+  if (step <= 0) {
+    // TODO: support negative strides
+    throw std::runtime_error("slice step must be positive");
+  }
+  if (start < 0) {
+    start += sizes[dim];
+  }
+  if (end < 0) {
+    end += sizes[dim];
+  }
+  if (start < 0) {
+    start = 0;
+  } else if (start >= sizes[dim]) {
+    start = sizes[dim];
```

Comment:
So Python slicing allows you to specify start/end greater than the size at that dimension?

Reply:
Yeah.

```
>>> x = [0]
>>> x[10:]
[]
```